### PR TITLE
[MCC-177198] Add unit tests

### PR DESCRIPTION
**Pull request by chenghuang-mdsol**
add Unit tests.

### CODE REVIEW

File: Medidata.Cloud.Thermometer.UnitTests/ThermometerTests.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Medidata.Cloud.Thermometer.UnitTests
+{
+    [TestClass]
+    public class ThermometerTests
+    {
+
+        public void GetJson(int port, string url, string expectedContain)
```
Comment (junshao) [inline Medidata.Cloud.Thermometer.UnitTests/ThermometerTests.cs]: I think this method can be private. In addition, I suggest refactoring this method to return a JSON result, as the method name indicates. And move the `Assert` into individual `TestMethod` for better readabitliy. Like

```cs
private string GetJson(int port, string routePath) {
   // Blah
   return json;
}

[TestMethod]
public void TestBlah() {
    // Arrange
    // Act
    var result = GetJson(port, routePath);

    // Assert
    Assert.Blah();
}

```

File: Medidata.Cloud.Thermometer.UnitTests/ThermometerTests.cs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Medidata.Cloud.Thermometer.UnitTests
+{
+    [TestClass]
+    public class ThermometerTests
```
Comment (junshao) [inline Medidata.Cloud.Thermometer.UnitTests/ThermometerTests.cs]: The name should be `ThermometerAppTests` to indicate the subject to test.

File: Medidata.Cloud.Thermometer.UnitTests/ThermometerTests.cs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            var app = new ThermometerApp();
+            app.Answer("howareyou", question => new { Answer = "I am fine. Thank you. And you?" });
+            Task.Run(() => app.Listen(9001)).Wait();
+            GetJson(9001, "/howareyou", "I am fine. Thank you. And you?");
+
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof (ArgumentException),"Question route path cannot be null or empty string.")]
+        public void TestEmptyPath()
```
Comment (junshao) [inline Medidata.Cloud.Thermometer.UnitTests/ThermometerTests.cs]: I don't think the "Test" suffix is required. And the name is better to indicate the expectation, like "EmptyRoutePathThrowsException", "NormalRoutePathReturnJson" and so forth.

**Comment by junshao** [comment]
The PR title requires to contain the MCC number, so that JIRA can sense the association if this PR was merged.

Title should be like `[MCC-177198] Add unit tests`, because you can find the MCC in JIRA
image

**Comment by art2003** [comment]
Can we add expectation in the method name as well, easy to read?
Example: TestBadPath_Whenxxxx_Thenxxxx

**Comment by junshao** [comment]
Closing this PR as it has merging conflicts and needs change. @chenghuang-mdsol Can you please modify the unit tests and resend a PR? Thanks.

**Pull request closed without merging**